src/databricks/labs/dqx/llm/README.md

```diff
@@ -0,0 +1,397 @@
+# 🤖 LLM-Based Primary Key Detection for DQX
+
+This module provides **optional** LLM-based primary key detection capabilities for the DQX data quality framework. The functionality is completely optional and only activates when users explicitly request it.
+
+## 🎯 **Overview**
+
+The LLM-based Primary Key Detection uses Large Language Models (via DSPy and Databricks Model Serving) to intelligently identify primary keys from table schema and metadata. This enhances the DQX profiling process by automatically detecting primary keys and generating appropriate uniqueness validation rules.
+
+## ✅ **Key Features**
+
+- **🔧 Completely Optional**: Works without any LLM dependencies by default
+- **🤖 Intelligent Detection**: Uses LLM analysis of table schema and metadata
+- **✨ Multiple Activation Methods**: Various ways to enable when needed
+- **🛡️ Graceful Fallback**: Clear messaging when dependencies unavailable
+- **⚡ Performance Optimized**: Lazy loading and conditional execution
+- **🔍 Duplicate Validation**: Optionally validates detected PKs for duplicates
+- **📊 Confidence Scoring**: Provides confidence levels and reasoning
+- **🔄 Retry Logic**: Handles cases where initial detection finds duplicates
+
+## 📦 **Installation**
+
+### **Regular DQX Usage (No LLM)**
+```bash
+# Standard installation - no additional dependencies needed
+pip install databricks-labs-dqx
+
+# Use normally - LLM features remain dormant
+from databricks.labs.dqx.config import ProfilerConfig
+config = ProfilerConfig()  # Works out of the box!
+```
+
+### **LLM-Enhanced Usage**
+```bash
+# Install DQX with LLM dependencies
+pip install databricks-labs-dqx
+pip install dspy-ai databricks_langchain
+
+# Now you can enable LLM features when needed
+config = ProfilerConfig(enable_llm_pk_detection=True)
+```
+
+## 🚀 **Usage Examples**
+
+### **Method 1: Configuration-Based**
+```python
+from databricks.labs.dqx.config import ProfilerConfig
+from databricks.labs.dqx.profiler.runner import ProfilerRunner
+
+# Enable LLM-based PK detection via configuration
+config = ProfilerConfig(
```

Comment:
as a first example i would specify just ProfilerConfig(). Then show how to customize.